cmd/virt-launcher/virt-launcher.go

```diff
@@ -610,15 +611,21 @@ func RemoveContents(dir string) error {
 
 func terminateIstioProxy() {
 	if istioProxyPresent() {
-		resp, err := http.Post(fmt.Sprintf("http://localhost:%d/quitquitquit", infraconfigurators.EnvoyMergedPrometheusTelemetryPort), "", nil)
+		httpClient := &http.Client{
+			Timeout: httpRequestTimeout,
+		}
+		resp, err := httpClient.Post(fmt.Sprintf("http://localhost:%d/quitquitquit", infraconfigurators.EnvoyMergedPrometheusTelemetryPort), "", nil)
 		if err != nil || resp.StatusCode != http.StatusOK {
 			log.Log.Error("Failed to request Istio proxy termination")
 		}
 	}
 }
 
 func istioProxyPresent() bool {
-	resp, err := http.Get(fmt.Sprintf("http://localhost:%d/healthz/ready", infraconfigurators.EnvoyHealthCheckPort))
+	httpClient := &http.Client{
+		Timeout: httpRequestTimeout,
+	}
+	resp, err := httpClient.Get(fmt.Sprintf("http://localhost:%d/healthz/ready", infraconfigurators.EnvoyHealthCheckPort))
 	if err != nil {
 		return false
```

Comment:
Logging the error can be useful here, especially if this is indeed hangs.

Reply:
Done